### PR TITLE
Feature/nil queue

### PR DESCRIPTION
**Pull request by tschoppi**
The behavior of instantiating a Worker with only `[nil]` as queue was ill-defined; The worker would allow working on all unqueued jobs, as well as on the first queued job that appears in the results of the poller. This is of course unacceptable.

This set of commits aims at reducing any uncertainty regarding what queues are respected.

### CODE REVIEW

File: lib/workhorse/poller.rb [hunk 1]
```diff
@@ -98,11 +98,13 @@ def queued_db_jobs(limit)
       select = select.where(table[:queue].not_in(bad_queries_select))
 
       # Restrict queues to "open" ones
-      if worker.queues.any?
+      unless worker.queues.empty?
         where = table[:queue].in(worker.queues.reject(&:nil?))
+
         if worker.queues.include?(nil)
           where = where.or(table[:queue].eq(nil))
```
Comment (tschoppi) [inline lib/workhorse/poller.rb]: The SQL generated here if only `[nil]` is passed as queue is:
```sql
WHERE (1=0 OR `jobs`.`queue` IS NULL)
```

**Comment by tschoppi** [comment]
Maybe some more involved examples like the one in this use case could be included in the FAQ.

File: lib/workhorse/poller.rb (hunk 1 above)
Comment (remofritzsche) [reply lib/workhorse/poller.rb]: Where does the ‚1=0‘ come from? I think it would be nice if we could work around that.

Comment (remofritzsche) [reply lib/workhorse/poller.rb]: Now I get it: The empty *where in*. I think we should omit the *where in* entirely if only *nil* is contained.

**Pull request merged**